src/Engine/UnitMods/DirectDamageModifier.java

```diff
@@ -0,0 +1,21 @@
+package Engine.UnitMods;
+
+import Engine.Combat.StrikeParams;
+
+public class DirectDamageModifier implements UnitModifierWithDefaults
+{
+  private static final long serialVersionUID = 1L;
+  private int attackModifier = 0;
+
+  public DirectDamageModifier(int firepowerChange)
+  {
+    attackModifier = firepowerChange;
+  }
+
+  @Override
+  public void modifyUnitAttack(StrikeParams params)
+  {
+    if( params.battleRange < 2 )
+      params.attackPower += attackModifier;
```

Comment:
Does AW1 use both multipliers and modifiers in its damage calculations?

Reply:
Technically, literally everything is multiplicative except for COP stats overwriting normal stats.
Since additive buffs work just fine for that case and the mathematical result should be identical (modulo rounding error maybe), IMO it makes most sense to use this value for Max.
Max's COP stats are like 150 (base) +10 (power boost) and then multiply that all by 1.1x.
I probably could have used division defense for Sami (in the next PR), now that I think about it, but I gave her inf the Kanbei treatment (replace D2D buff with COP buff) since I didn't.

Rounding error is likely beyond what I can find it in myself to care about.

Reply:
Sure, for Max you could use either this or a `DamageMultiplierOffense` of 1.2, and the combat math should work out the same, yeah? Though if literally everything is multiplicative except for stat overwriting, I question that we have multiple methods of stat augmentation.

Reply:
You are missing that `DamageMultiplierOffense` *multiplies* by 1.2, which would also affect his D2D buff. (and the generic COP buff of 1.1x scales both)
I don't want to have things adding to the damage multiplier because that creates order-dependence for your buffs.
(The "downside" of this is that a few COs get bigger power buffs for tag partners than they "should", but I'm not too torn up about it since AW1 COs are mostly weak and balance in tags is going to not exist anyway.)

Everything (in AW1) is multiplicative aside from COP stats replacing D2D stats.

Reply:
> You are missing that `DamageMultiplierOffense` multiplies by 1.2, which would also affect his D2D buff

Ok, just to lay this out more plainly to make sure I understand. The D2D is +50 for a base attack power of 150. Max force adds a base multiplier of 1.1 plus a direct boost of 20.
150 * 1.1 + 20 = 185 power. Whereas if we used a multiplier here it would be
150 * 1.1 * 1.2 = 198 power. Right?
I guess the obvious question; why not represent the AW1 base buffs as damage modifiers instead of multipliers to avoid order dependence and narrow to having one right and obvious way to do things?

> Everything (in AW1) is multiplicative aside from COP stats replacing D2D stats

Does that mean the D2D should not be active while the COP is?

Reply:
> The D2D is +50 for a base attack power of 150.
> Max force adds a base multiplier of 1.1 plus a direct boost of 20.
> 150 * 1.1 + 20 = 185 power.

It's actually (150 + 20) * 1.1 = 187 power, but yes.
> Whereas if we used a multiplier here it would be 150 * 1.1 * 1.2 = 198 power.

Yes.
> Right? I guess the obvious question; why not represent the AW1 base buffs as damage modifiers instead of multipliers to avoid order dependence and narrow to having one right and obvious way to do things?

Because that doesn't mathematically work. I can't add one constant number to the attack value for Andy and Max to get the right buff amount - it's 1.1x whatever the stat is.
Also, I can't even add one constant number to Eagle, Drake, and (next PR) Sami because all 3 of them have different statlines for different units.
The buff has to be a multiplier because that provides the correct result (and is also order-independent, as long as you group the mutually-commutative operations).
> Does that mean the D2D should not be active while the COP is?

I get around that by using addition, since that works for attack.
This also lets me use the same buff type that I'll use for the other 3 Max variants, and it gives correct results in all cases.

Kanbei does work like this (replacing D2D with powers) because he subtracts 10 from his defensive multiplier and "multiplies damage taken by 7/8ths" is a super janky thing to put in a power description.

Reply:
Okay, it sounds like the answer to my initial question was a simple "yes". Is Kanbei the only one who deactivates his D2D during COP? and that matches cart behavior? That's pretty strange.

Reply:
In my implementation, yes.
The original uses tables of stats which multiply with the CO's global attack and defense multipliers (which Kanbei and Sturm notably modify during D2D).

Kanbei deactivates his D2D during COP to replace it with the stronger COP stats, because math for 0.8x to 0.7x would be weird.
It might be doable with DoR-style defense, but I didn't think of that.

Sturm and Sami will also have similar effects:
Sturm (like Kanbei) has a global stat multiplier that he makes integer changes to.
Sami has an infantry attack/defense boost that she makes integer changes to (in addition to the default power boost).

It'd be possible to encode their final effects explicitly, but I like modeling the math in a similar way to the original.
The reason it's not exactly like the original is I don't want to mix multiplication and addition on a single value because that makes order important.